tests/geometry/functional/test_so3.py

```diff
@@ -41,7 +54,7 @@ def test_inverse(batch_size: int, dtype: torch.dtype):
     group = so3.rand(batch_size, generator=rng, dtype=dtype)
 
     # check analytic backward for the operator
-    check_lie_group_function(so3, "inverse", TEST_EPS, group)
+    check_lie_group_function(so3, "inverse", TEST_EPS, (group,))
```

Comment:
You can make all this code much more concise and readable, and all in a single unit tests, by doing something similar to:


```python
FN_CFGS = [
  ("inverse", ("group0",)),
  ("hat", ("tangent",)),
  ("compose", ("group0", "group1")),
]
@pytest.mark.parametrize(fn_cfg, FN_CFGS)
def test_so3_op(fn_cfg):
  # create the generator, manual seed, etc. (also include batch size, etc.)
  input_vals = {
    "group0": so3.rand(batch_size, generator=rng, dtype=dtype), 
    "tangent": torch.rand(batch_size, 3, dtype=dtype, generator=rng), 
    "group1": torch.rand(batch_size, 3, dtype=dtype, generator=rng),
    "matrix": torch.rand(
        batch_size,
        int(torch.randint(1, 20, (1,), generator=rng)),
        3,
        dtype=dtype,
        generator=rng,
    ),
  }  # and anything that's useful like near_zero stuff or w/e

  fn_inputs = tuple(input_vals[k] for k in fn_cfg[1])   # convert cfg to tensors
  check_lie_group_function(so3, fn_cfg[0], TEST_EPS, fn_inputs)  
```